iothub/device/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -37,25 +33,6 @@ public sealed class IotHubConnectionStringBuilder
 
         private const string CommonX509CertPropertyName = "x509";
 
-        private const RegexOptions CommonRegexOptions = RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant;
```

Comment:
I'm removing the regex checks in here, since these character restrictions could be updated by service at any time, and we would be required to keep them updated. Instead, we'll only check the following:
- device Id is set
- IoT hub hostname is set
- the shared access key supplied is a valid base64 string
- the shred access signature supplied follows the expected format